hadoop-gremlin/pom.xml

```diff
@@ -219,6 +219,35 @@ limitations under the License.
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
             </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>check-hadoop-installed</id>
```

Comment:
Could you clarify the need for enforcer for these environment variables? Was the build failing without those somehow? I don't have them set at the moment and `mvn clean install` builds fine for me.

Reply:
My idea is to check the environment the build/tests run in and warn that it is incomplete, instead of throwing cryptic exceptions. Having this stated in the POM serves also as some documentation. In this specific case I could not realize the needed build environment setup neither how the CI environment provides Hadoop libraries for the tests

Reply:
I see where not setting `HADOOP_GREMLIN_LIBS` generates some WARN messages when you build:

```text
[WARN] org.apache.tinkerpop.gremlin.hadoop.jsr223.HadoopGremlinPlugin - Be sure to set the environmental variable: HADOOP_GREMLIN_LIBS
   [WARN] org.apache.hadoop.util.NativeCodeLoader - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
   [WARN] org.apache.tinkerpop.gremlin.hadoop.jsr223.HadoopGremlinPlugin - Be sure to set the environmental variable: HADOOP_GREMLIN_LIBS
   [WARN] org.apache.tinkerpop.gremlin.hadoop.jsr223.HadoopGremlinPlugin - Be sure to set the environmental variable: HADOOP_GREMLIN_LIBS
   [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.844 s - in org.apache.tinkerpop.gremlin.hadoop.jsr223.HadoopGremlinPluginTest
```

but it doesn't error, even when you run integration tests on `hadoop-gremlin`. I think those environment variables are really more for usage at runtime rather than for actual tests. You mentioned that you get "cryptic exceptions" - what are those exceptions and do they occur during build on windows?

I'd rather not force folks to setup up environment variables unless they are necessary to the build somehow. But, now that I look at it again, i assume that `<level>WARN</warn>` isn't actually enforcement (meaning the build doesn't fail if those environment variables aren't present). So perhaps this addition is ok given how you have done it even if there aren't errors specifically.